src/main/java/app/sami/languageWeb/contract/ContractController.java

```diff
@@ -0,0 +1,72 @@
+package app.sami.languageWeb.contract;
+
+import app.sami.languageWeb.contract.dtos.*;
+import app.sami.languageWeb.contract.models.Contract;
+import app.sami.languageWeb.contract.models.Status;
+import app.sami.languageWeb.spring.binds.RequestJwtSubject;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@AllArgsConstructor
+public class ContractController {
+
+    private final ContractRepository contractRepository;
+    private final ContractService contractService;
+
+    @PostMapping("/contracts/storage/upload-uri")
+    public StorageUriDto getUploadUri(@RequestJwtSubject UUID subject,
+                                      @RequestBody String fileName){
+        return contractService.getUploadUri(subject, fileName);
+    }
+
+    @PutMapping("/contracts/{contractId}/file")
+    public ContractDto addTranslatedDocument(@RequestJwtSubject UUID subject,
+                                             @PathVariable Long contractId,
+                                             @RequestBody String path){
+        return ContractMapper.toContractDto(contractService.changeContractFile(subject, contractId, path));
+    }
+
+    @PutMapping("/contracts/{contractId}/extension")
```

Comment:
this should simply be `PUT /contracts/{contractId}` with body `ContractRequest` we rarely see, if ever, granular changes of entities

Reply:
here, you must check that the subject IS the request owner